### PR TITLE
Add Dockerfile for image build

### PR DESCRIPTION
**Pull request by markmcclain**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+RUN mkdir -p /data ; chown -R devpi:devpi /data
+VOLUME /data
+
+RUN apk add --no-cache tini su-exec bash ca-certificates \
+    && update-ca-certificates
+
+RUN apk add --no-cache --virtual .build-deps gcc python3-dev libffi-dev musl-dev \
+    && pip install --no-cache-dir --disable-pip-version-check \
+    devpi-server==$DEVPI_SERVER_VERSION \
```
Comment (7AC) [inline Dockerfile]: Don't we want to install the version from this GitHub repo rather than what's in pypi?

File: docker-entrypoint.sh [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+        export -f initialize_devpi
+        su-exec devpi bash -c initialize_devpi
+	unset initialize_devpi
+    fi
+
+    echo "[RUN]: Launching devpi-server"
+    exec su-exec devpi devpi-server --restrict-modify root --host 0.0.0.0 --port $DEVPI_PORT 
+fi
+
+echo "[RUN]: Builtin command not provided [devpi]"
```
Comment (7AC) [inline docker-entrypoint.sh]: Do we need this?

File: Dockerfile (hunk 1 above)
Comment (markmcclain) [reply Dockerfile]: yeah... good point. I've updated to use the local sources.

File: Dockerfile [hunk 3]
```diff
@@ -17,12 +12,14 @@ VOLUME /data
...
+ADD ./server /src/server
+ADD ./client /src/client
+ADD ./web /src/web
+
+RUN apk add --no-cache --virtual .build-deps gcc libffi-dev musl-dev \
+    && pip install --no-cache-dir --disable-pip-version-check /src/server/. \
+    && pip install --no-cache-dir --disable-pip-version-check /src/client/. \
+    && pip install --no-cache-dir --disable-pip-version-check /src/web/. \
     && apk del .build-deps \
     && rm -r /root/.cache
```
Comment (7AC) [inline Dockerfile:24]: && rm -r /src ?

File: docker-entrypoint.sh [hunk 4]
```diff
@@ -30,14 +30,13 @@ if [ "$1" = 'devpi' ]; then
     if [ ! -f  $DEVPISERVER_SERVERDIR/.serverversion ]; then
         export -f initialize_devpi
         su-exec devpi bash -c initialize_devpi
-	unset initialize_devpi
+        unset initialize_devpi
     fi
 
     echo "[RUN]: Launching devpi-server"
```
Comment (7AC) [inline docker-entrypoint.sh:36]: I'm not sure about all these echo, the services themselves don't emit anything?

File: Dockerfile (hunk 3 above)
Comment (markmcclain) [reply Dockerfile:24]: fixed

**Pull request merged**